road/validation_road.py

```diff
@@ -0,0 +1,113 @@
+import os
+import pandas as pd
+from simpledbf import Dbf5
+import configparser
+import json
+import yaml
+from sklearn.linear_model import LinearRegression
+from io import StringIO
+import geopandas as gpd
+from scatter import *
+
+
+config = configparser.ConfigParser()
+config.read('config.ini')
+
+# Extract the CHAMP input file names
+dbf_directory = config['DBF']['DBF_Directory']
+dbf_files = [dbf_file.strip() for dbf_file in config['DBF']['DBF_Files'].split(',')]
+dbf_column_names = [name.strip() for name in config['DBF']['DBF_Column_Names'].split(',')]
+
+# Extract the observed file name and tab
+excel_file_path = config['OBSERVED FILE']['Excel_File_Path']
+sheet_name = config['OBSERVED FILE']['Sheet_Name']
+extra_columns = [col.strip() for col in config['OBSERVED FILE']['Excel_Extra_Columns'].split(',')]
+obs_usecols = config['OBSERVED FILE']['Obs_usecols']
+time_periods = ['AM', 'MD', 'PM', 'EV', 'EA']
+
+#Part 1 
+# Scatter variables
+chosen_timeperiod = config['SCATTER INPUT']['Chosen_period']
+classification_col = config['SCATTER INPUT']['Classification_col']
+combined_df_cols = [col.strip() for col in config['SCATTER INPUT']['Combined_DF_Cols'].split(',')]
+
+# Est-Obs Plot
+fields1 = config['EST SCATTER PLOT']['Fields'].split(', ')
+nominal_fields1 = config['EST SCATTER PLOT']['NominalFields'].split(', ')
+x_field1 = config['EST SCATTER PLOT']['XField']
+y_field1 = config['EST SCATTER PLOT']['YField']
+name1 = config['EST SCATTER PLOT']['Name']
+
+# Percentile Plot
+fields2 = config['Perc SCATTER PLOT']['Fields'].split(', ')
+nominal_fields2 = config['Perc SCATTER PLOT']['NominalFields'].split(', ')
+x_field2 = config['Perc SCATTER PLOT']['XField']
+y_field2 = config['Perc SCATTER PLOT']['YField']
+name2 = config['Perc SCATTER PLOT']['Name']
+
+dashboard_num = config['YAML']['Dashboard_number']
+
+
+
+def handle_merged_header(excel_file, sheet_name, extra_columns):
+    df = pd.read_excel(excel_file, sheet_name=sheet_name, header=None, skiprows=1)
+    header = df.iloc[0]
+    df = df[1:]
+    df.columns = header
+    return df[extra_columns]
+
+def filter_and_aggregate(excel_file, sheet_name, dbf_directory, dbf_files, column_names, time_periods, extra_columns):
+    excel_df = handle_merged_header(excel_file, sheet_name, extra_columns)
+    base_df = excel_df.reset_index(drop=True)
+
+    for col in time_periods + ['Daily']:
+        base_df[col] = 0
+
+    for dbf_file, output_column in zip(dbf_files, time_periods):
+        dbf_file_path = f"{dbf_directory}\{dbf_file}"
+        dbf = Dbf5(dbf_file_path)

```

Comment:
question on the DBF file readings: what are these? Can you verify if e.g. geopandas cannot read these dbfs in correctly? If it can, that will reduce the need to install all the extra packages and conversion to dataframes, since geopandas is a pretty standard package in geo/dataframes analysis work at this point

Reply:
geopandas takes 3.5 min to read a single dbf file, but Dbf5 only needs 7 seconds

Reply:
woah that's a huge difference! do you have a general sense why this might be the case?

Reply:
I guess Dbf5 might read the DBF file in a more straightforward manner, whereas GeoPandas could be preparing the data for further geospatial operations.

Reply:
right. Let's stick with Dbf5 for now then I guess.